Apps/WebClient/src/ClientApp/app/router.ts

```diff
@@ -1,63 +1,107 @@
+import ImmunizationsComponent from '../app/views/immunizations.vue'
+import NotFoundComponent from '../app/views/errors/notFound.vue'
+import LoginComponent from '../app/views/login.vue'
+import LogoutComponent from '../app/views/logout.vue'
+import UnauthorizedComponent from '../app/views/errors/unauthorized.vue'
+import TempAuthComponent from '../app/views/auth.vue'
 
 Vue.use(VueRouter)
 
+function load(componentPath: string) {
+    // '@' is aliased to src/components, so we can't use it with our structure
+    return () => import(`./views/${componentPath}`)
+}
+
 const routes = [
-    { path: '/', component: LandingComponent },
-    { path: '/registration', component: RegistrationComponent },
     {
+        path: '/',
+        component: LandingComponent,
+        meta: { requiresAuth: false }
+    },
+    {
+        path: '/registration',
+        component: RegistrationComponent,
+        meta: { requiresAuth: false }
+    },
+    {
+        name: 'home',
         path: '/home',
         component: HomeComponent,
         meta: { requiresAuth: true, roles: ['user'] },
         children: [
-            { path: '/immunizations', component: ImmunizationsComponent, meta: { requiresAuth: true, roles: ['user'] } }
+            {
+                path: '/immunizations',
+                component: ImmunizationsComponent,
+                meta: { requiresAuth: true, roles: ['user'] }
+            }
         ]
     },
+    {
+        path: '/auth',
+        component: TempAuthComponent,
+        meta: { requiresAuth: false }
+    },
+    {
+        path: '/login',
+        component: LoginComponent,
+        meta: { requiresAuth: false, roles: ['user'] },
+    },
     {
         path: '/logout',
         component: LogoutComponent,
-        meta: { requiresAuth: true, roles: ['user'] }
+        meta: { requiresAuth: true, roles: ['user'] },
     },
-    { path: '/unauthorized', component: UnauthorizedComponent }, // Unauthorized
+    {
+        path: '/unauthorized',
+        component: UnauthorizedComponent,
+        meta: { requiresAuth: false }
+    }, // Unauthorized
     { path: '*', component: NotFoundComponent } // Not found; Will catch all other paths not covered previously
 ]
 
+
 const router = new VueRouter({
     mode: 'history',
     routes
 })
 
-router.beforeEach((to, from, next) => {
+router.beforeEach(async (to, from, next) => {
     if (to.meta.requiresAuth) {
-        //const auth = store.state.security.auth
-        const auth = { authenticated: false };//store.state.security.auth
+        let isAuthenticated = store.getters.isAuthenticated;
         console.log('Requires auth!');
-        if (!auth.authenticated) {
+        if (!isAuthenticated) {
             //security.init(next, to.meta.roles)
             next({ name: 'unauthenticated' })
             console.log('Not authenticated');
+            console.log(to.path);
+            // commit the to.path to vuex here
+            store.commit('requestedRoute', to.path);
```

Comment:
Idea, instead of doing a state commit here, what if we pass the to.path as a query string to the Login Component. That way we avoid changing the state until the components stats the login flow

Reply:
let's do this later.. I'll get a successfully PR build going now wit the fixed msising npm package